modules/data_integrity_flag/php/data_integrity_flag.class.inc

```diff
@@ -127,6 +129,17 @@ class Data_Integrity_Flag extends \NDB_Menu_Filter
         $this->order_by     = 'dataflag_date desc';
         $this->validFilters = [];
         $this->formToFilter = [];
+
+        $users         = array();
+        $selectedUsers = $db->pselect(
+            "SELECT u.UserID, u.Real_name FROM data_integrity_flag dif " .
+            "INNER JOIN users u ON (dif.dataflag_userid=u.UserID)",
+            array()
+        );
+        foreach ($selectedUsers as $row) {
+            $users[$row['UserID']] = $row['Real_name'];
+        }
```

Comment:
seems like you're trying to do a `pselectWithIndexKey`?

Reply:
@driusan the problem with pselectWithIndexKey in this case is that i need to query with a 'SELECT DISTINCT'. do you prefer i do this or just turn the filter field from a select to a textbox?